api/v1alpha1/stack_type.go

```diff
@@ -4,122 +4,125 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-type Tinkerbell struct {
+type Stack struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// Spec describes the desired tinkerbell stack state.
-	Spec TinkerbellStackSpec `json:"spec,omitempty"`
+	Spec StackSpec `json:"spec"`
 }
 
-// TinkerbellStackSpec specifies details of an Tinkerbell setup.
-type TinkerbellStackSpec struct {
+// StackSpec specifies details of an Tinkerbell setup.
+type StackSpec struct {
 	// Version is the Tinkerbell CRD version.
 	Version string `json:"version"`
 
 	// DNSResolverIP is indicative of the resolver IP utilized for setting up the nginx server responsible for proxying
 	// to the Tinkerbell service and serving the Hook artifacts.
-	DNSResolverIP string `json:"dnsResolverIP"`
+	DNSResolverIP string `json:"dnsResolverIP,omitempty"`
 
 	// Registry is the registry to use for all images. If this field is set, all tink service deployment images
 	// will be prefixed with this value. For example if the value here was set to docker.io, then boots image will be
 	// docker.io/tinkerbell/boots.
-	Registry string `json:"registry"`
+	Registry string `json:"registry,omitempty"`
 
-	// ImagePullSecret the secret name containing the docker auth config which should exist in the same namespace where
+	// ImagePullSecrets the secret name containing the docker auth config which should exist in the same namespace where
 	// the operator is deployed(typically tinkerbell)
-	ImagePullSecret []string `json:"imagePullSecret"`
+	ImagePullSecrets []string `json:"imagePullSecrets,omitempty"`
 
 	// Services contains all Tinkerbell Stack services.
-	Services Services `json:"services"`
+	Services Services `json:"services,omitempty"`
 }
 
 // Services contains all Tinkerbell Stack services.
 type Services struct {
 	// Boots contains all the information and spec about boots.
-	Boots Boots `json:"boots"`
+	Boots Boots `json:"boots,omitempty"`
 
 	// Hegel contains all the information and spec about boots.
-	Hegel Hegel `json:"hegel"`
+	Hegel Hegel `json:"hegel,omitempty"`
 
 	// Rufio contains all the information and spec about rufio.
-	Rufio Rufio `json:"rufio"`
+	Rufio Rufio `json:"rufio,omitempty"`
 
 	// TinkServer contains all the information and spec about tink server.
-	TinkServer TinkServer `json:"tinkServer"`
+	TinkServer TinkServer `json:"tinkServer,omitempty"`
 
 	// TinkController contains all the information and spec about tink controller.
-	TinkController Image `json:"tinkController"`
+	TinkController TinkController `json:"tinkController,omitempty"`
 }
 
 // Boots specifies the deployment details of Tinkerbell service, Boots.
 type Boots struct {
 	// Image specifies the details of a tinkerbell services images
-	Image Image `json:"image"`
+	Image Image `json:"image,omitempty"`
 
 	// DHCPAddressListener set the ip and port to listen on for DHCP.
-	DHCPAddressListener string `json:"dhcpAddress"`
+	DHCPAddressListener string `json:"dhcpAddressListener,omitempty"`
```

Comment:
Why `DHCPAddressListener` vs `DHCPAddress`? The `Address` appendage aligns with other pieces in the API. 

I do appreciate there may be underlying configuration but that's changing with a recent PR and boots has historically had quirky arguments so I wouldn't follow them.

Reply:
We discussed this in the previous PR and this name is what we agreed to choose.

Reply:
@jacobweinstock What's the rationale? What value does 'listener' have? Why is does this have 'listener' but HTTP doesn't? I vote to just use `*Address` but if both of you prefer to have `Listener` or `Bind` that's fine, can we just be consistent?

Reply:
FWIW we use `-http-addr` in Boots now... https://github.com/tinkerbell/boots/blob/d135982a3767b050eea20e04615c76b0ef09e0b6/cmd/boots/flag.go#L99

Reply:
tbh I don't have a strong opinion about whether it should be ` listener/bind` or `address`. WDYT @jacobweinstock ?